### PR TITLE
Minor cleanups for Legacy Storyboard/Beatmap decoder

### PR DESCRIPTION
**Pull request by turbedi**
Mostly cosmetical, except for the changes in `LegacyStoryboardDecoder.handleEvents()`.
This way we only call `Substring` once at the end of the loop and not after every iteration. I will test the actual difference in a few minutes.

### CODE REVIEW

File: osu.Game/Beatmaps/Formats/LegacyBeatmapDecoder.cs [hunk 1]
```diff
@@ -300,13 +300,13 @@ private void handleEvent(string line)
             switch (type)
             {
                 case LegacyEventType.Background:
-                    string bgFilename = split[2].Trim('"');
-                    beatmap.BeatmapInfo.Metadata.BackgroundFile = bgFilename.ToStandardisedPath();
+                    string bgFilename = split[2];
+                    beatmap.BeatmapInfo.Metadata.BackgroundFile = CleanFilename(bgFilename);
```
Comment (smoogipoo) [inline osu.Game/Beatmaps/Formats/LegacyBeatmapDecoder.cs]: Put onto one line

File: osu.Game/Beatmaps/Formats/LegacyBeatmapDecoder.cs [hunk 2]
```diff
@@ -300,13 +300,13 @@ private void handleEvent(string line)
...
                     break;
 
                 case LegacyEventType.Video:
-                    string videoFilename = split[2].Trim('"');
-                    beatmap.BeatmapInfo.Metadata.VideoFile = videoFilename.ToStandardisedPath();
+                    string videoFilename = split[2];
+                    beatmap.BeatmapInfo.Metadata.VideoFile = CleanFilename(videoFilename);
```
Comment (smoogipoo) [inline osu.Game/Beatmaps/Formats/LegacyBeatmapDecoder.cs]: Same here

File: osu.Game/Beatmaps/Formats/LegacyStoryboardDecoder.cs [hunk 3]
```diff
@@ -65,12 +64,14 @@ protected override void ParseLine(Storyboard storyboard, Section section, string
...
+            var lineSpan = line.AsSpan();
 
-            while (line.StartsWith(" ", StringComparison.Ordinal) || line.StartsWith("_", StringComparison.Ordinal))
+            while (lineSpan.StartsWith(" ", StringComparison.Ordinal) || lineSpan.StartsWith("_", StringComparison.Ordinal))
             {
+                lineSpan = lineSpan.Slice(1);
                 ++depth;
-                line = line.Substring(1);
             }
+            line = line.Substring(depth);
```
Comment (smoogipoo) [inline osu.Game/Beatmaps/Formats/LegacyStoryboardDecoder.cs]: Maybe just `lineSpan.ToString()`.

**Review by smoogipoo — CHANGES_REQUESTED** [review]

**Comment by peppy** [comment]
@turbedi make sure to also [make inspectcode happy](https://ci.appveyor.com/project/peppy/osu/builds/30312143/messages).

File: osu.Game/Beatmaps/Formats/LegacyStoryboardDecoder.cs (hunk 3 above)
Comment (turbedi) [reply osu.Game/Beatmaps/Formats/LegacyStoryboardDecoder.cs]: I tried both, and using `Substring` seems to be a bit faster (difference is within margin of error though). Should I still change to `lineSpan.ToString()`?

**Comment by turbedi** [comment]
I tested it with the Storyboard of this beatmap: https://osu.ppy.sh/beatmapsets/470977#osu/1006822
Allocations went from 430.54 MB to 418.15 MB.

File: osu.Game/Beatmaps/Formats/LegacyStoryboardDecoder.cs (hunk 3 above)
Comment (bdach) [reply osu.Game/Beatmaps/Formats/LegacyStoryboardDecoder.cs]: If `Substring` doesn't have a significant performance advantage over even thousands of executions I'd change for the sake of readability.

**Review by peppy — APPROVED** [review]

**Review by peppy — APPROVED** [review]

**Pull request merged**